src/app/utils/field-validation/validators/numberValidator.ts

```diff
@@ -0,0 +1,75 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { flow } from 'fp-ts/lib/function'
+
+import { ProcessedSingleAnswerResponse } from 'src/app/modules/submission/submission.types'
+import { INumberField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+
+import { NumberSelectedValidation } from '../../../../types/field'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type NumberValidator = ResponseValidator<ProcessedSingleAnswerResponse>
+type NumberValidatorConstructor = (numberField: INumberField) => NumberValidator
+
+/**
+ * Validator, to check that answer is an empty string or a nonnegative integer
```

Comment:
nit: I find it useful to start function docs with a verb, and also to try not to repeat information which is already apparent from the function name
```suggestion
 * Checks that answer is an empty string or a nonnegative integer
```